templates/cluster-template-ssh-material.yaml

```diff
@@ -8,47 +8,50 @@ spec:
     pods:
       cidrBlocks:
         - 192.168.0.0/16
-    serviceDomain: cluster.local
+    serviceDomain: "cluster.local"
   infrastructureRef:
-    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
     kind: CloudStackCluster
     name: ${CLUSTER_NAME}
   controlPlaneRef:
     kind: KubeadmControlPlane
     apiVersion: controlplane.cluster.x-k8s.io/v1beta1
     name: ${CLUSTER_NAME}-control-plane
 ---
-apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
 kind: CloudStackCluster
 metadata:
   name: ${CLUSTER_NAME}
 spec:
   controlPlaneEndpoint:
     host: ${CLUSTER_ENDPOINT_IP}
-    port: ${CLUSTER_ENDPOINT_PORT}
-  zones:
-  - name : ${CLOUDSTACK_ZONE_NAME}
-    network: 
-      name: ${CLOUDSTACK_NETWORK_NAME}
+    port: ${CLUSTER_ENDPOINT_PORT=6443}
+  failureDomains:
+    - name: ${CLOUDSTACK_FD1_NAME=failure-domain-1}
+      acsEndpoint:
+        name: ${CLOUDSTACK_FD1_SECRET_NAME}
+        namespace: default
+      zone:
+        name :  ${CLOUDSTACK_ZONE_NAME}
+        network:
+          name: ${CLOUDSTACK_NETWORK_NAME}
 ---
-apiVersion: controlplane.cluster.x-k8s.io/v1beta1
 kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
 metadata:
   name: "${CLUSTER_NAME}-control-plane"
 spec:
   kubeadmConfigSpec:
-    clusterConfiguration:
-      imageRepository: k8s.gcr.io
```

Comment:
should we keep this?

Reply:
I'm not entirely sure of the implications of removing it, so I left it in from the preceding version.